eng/Build.props

```diff
@@ -102,9 +102,10 @@
                         Include="$(RepoRoot)src\Installers\Rpm\**\*.*proj" />
       </ItemGroup>
 
-      <ItemGroup Condition=" '$(TargetOsName)' == 'win' AND ('$(TargetArchitecture)' == 'x86' OR '$(TargetArchitecture)' == 'x64') ">
+      <ItemGroup Condition=" '$(TargetOsName)' == 'win' ">
         <NativeProjects Include="$(RepoRoot)src\**\*.vcxproj" Exclude="@(ProjectToExclude)" AdditionalProperties="Platform=x64" />
         <NativeProjects Include="$(RepoRoot)src\**\*.vcxproj" Exclude="@(ProjectToExclude)" AdditionalProperties="Platform=Win32" />
+        <NativeProjects Include="$(RepoRoot)src\**\*.vcxproj" Exclude="@(ProjectToExclude)" AdditionalProperties="Platform=arm64" />
```

Comment:
@HaoK this is probably the root of your problem. Instead, leave the `Condition` alone but add another `<ItemGroup />` specifically for `'$(TargetArchitecture)' == 'arm64'`

Reply:
Cool I'll try that, so will the PR you did in https://github.com/dotnet/aspnetcore/commit/d4d4c4bda5f36a8ecb7cc5c66ef4ad6ae72f1df5 that removed arm64 need to be rejiggered at all?  We'll want to build arm64 ANCM and try running those on arm64 helix queues eventually (once I get things strung together enough)

Reply:
The real issue here is the need to test ARM64-specific native binaries i.e. we don't **want** fan-out here ☹️

As we discussed in a meeting, we don't want to find a reason to create another build job just for a Windows ARM64 build that submits to a Windows ARM64 Helix queue, especially so soon after I got rid of the need for a separate Linux ARM64 build that submits to a couple of ARM64 queues. But I'm not sure how to avoid it since I don't think we have ARM64 build agents (we cross-compile for ARM64). (Am I correct all of the tests you're changing here run on Helix agents❔)

I can't think of a way around that ickiness, short of significant complications like another (ARM64-specific) build step and even-more ugly correlation payload creation. My main suggestion is to test _only_ the newly changed projects in the new "Helix Windows ARM64" jobs. Secondarily, those jobs should exist only in aspnetcore-helix-matrix and aspnetcore-quarantined-tests.

Reply:
Can we just do something like for the arm queues, we build using arm64 like we did before, and as a result that would be pushing the arm64 native binaries to the windows queue like we want.  

I guess I don't really care how we choose to implement this, we aren't adding any new tests per see, we will just be running the existing IIS functionals tests on arm64 now where we weren't building or running those tests before

Reply:
Hrm, for some reason putting the arm64 in its own item group results in a pack error, not sure why its trying to build arm64 when doing x64 builds, @dougbu any ideas why this works in the original group but not with your suggestion split out?

```
#[error].dotnet\sdk\7.0.100-alpha.1.21613.8\Sdks\NuGet.Build.Tasks.Pack\build\NuGet.Build.Tasks.Pack.targets(221,5): error : (NETCORE_ENGINEERING_TELEMETRY=Build) Could not find a part of the path 'D:\a\_work\1\s\artifacts\bin\AspNetCoreModuleShim\ARM64\Release'.
```

Reply:
Likely another `Condition` (or similar) is mixing x64 and ARM64. This was just the first one I noticed. General point is to build *and use* native bits for the ARM64 platform only when cross-compiling for that architecture.

Reply:
Thinking out loud: This might also have something to do w/ the updated test matrix.

Reply:
Any ideas what the best path forward is?  The previous iteration at least got to the tests running (albeit with the wrong bits)

I am able to build locally with things either way, so I'm not sure what the ci does differently, although I do notice that the arm64 build job doesn't seem to be producing the ANCM/arm64 dlls strangely since those do get generated when running build.cmd in the servers/IIS directory